routes/adjustmentMethods/ManualAdjustmentMethod.ts

```diff
@@ -0,0 +1,16 @@
+import { AdjustmentMethod, AdjustmentMethodResponse, AdjustmentOptions } from "./AdjustmentMethod";
+
+
+/**
+ * Only returns time data.
+ */
+async function calculateManualWateringScale( ): Promise< AdjustmentMethodResponse > {
+	return {
+		scale: -1
```

Comment:
I dont think this routine should set `scale` but rather "do nothing". See broader comment on use of "-1".

Reply:
Can you elaborate further on what it should return or how it should work?

Reply:
So the way Firmware works is to parse the response looking for the various tags (e.g. "scale"). If it doesn't see the tag then it doesn't update the variable. In the case of "scale", Firmware only uses values "0<value<250" and ignores any negative values. So not adding the variable will result in the same outcome as setting it to "-1". Unless we define a standard meaning for "-1" (e.g. use previous) then we shouldn't be using it.

Also, the purist in me thinks that given this is the Manual method then we shouldn't change the "scale" value as the users has said not to. I think the most we should do is set a 1h rain delay.

Reply:
Should we fix this by setting `scale` to `undefined` instead of `-1`?

Reply:
Not sure that either of those is right. Setting it to "undefined" feels wrong as `scale` is most definitely set to something, we just don't know what ! It also doesn't feel right to use "-1". I would suggest we only include fields in the response that we actively want to change. So in this case we would return an empty object.

Reply:
i.e. we should only return values to Firmware that we want to change and not things that we want to stay the same.

Reply:
We can explicitly set it to `undefined` so we don't need to make it optional in the `AdjustmentMethodResponse` and omit the property from the response when it is sent to the controller.

Reply:
So I'll admit to preferring "optional" and taking an incremental approach where things are dealt with as they come up rather than "all at the end" which is what we current have. I've been caught out too many times with the later approach and forgotten about something really important because it wasn't dealt with in context.

As an example, I was going to submit a PR that builds up the response incrementally and then only returns the values that have been updated:

```
export const getWateringData = async function( req: express.Request, res: express.Response ) {
...
	let result: WateringResponse			= {};
...
	// X-Forwarded-For header may contain more than one IP address and therefore
	// the string is split against a comma and the first value is selected
	result.eip = ipToInt( remoteAddress.split( "," )[ 0 ] );

...
	// Attempt to resolve provided location to GPS coordinates and timezone.
	let coordinates: GeoCoordinates;
	try {
		coordinates = await resolveCoordinates( location );
		let timeData: TimeData = getTimeData( coordinates );

		result.tz = getTimezone( timeData.timezone, undefined );
		result.sunrise = timeData.sunrise;
		result.sunset = timeData.sunset;
	} catch (err) {
		res.send(`Error: Unable to resolve location (${err})`);
		return;
	}
...
	// Return the response to the client in the requested format including only defined values
	if ( outputFormat === "json" ) {
		res.json( result );
	} else {
		res.send(
			( "scale" in result	? "&scale="	+ result.scale : "") +
			( "rd" in result	? "&rd=" + result.rd : "") +
			( "tz" in result	? "&tz=" +	result.tz : "") +
			( "sunrise" in result	? "&sunrise="	+ result.sunrise : "" ) +
			( "sunset" in result	? "&sunset=" + result.sunset : "" ) +
			( "eip" in result	? "&eip=" + result.eip : "" ) +
			( "rawData" in result	? "&rawData=" + JSON.stringify( result.rawData ) : "" )
		);
	}

```
Probably more of a style things so not going to die in a ditch over this on 8)

Reply:
I'm personally against using "monolithic" objects because I think it doesn't play nicely with typing. Since not all the properties in the object are initialized when it is created, they must all be declared as optional. Making the fields optional increases the chances that somebody will accidentally forget to set one, and makes it unclear what kind of data a function will be passed (i.e. will the object contain just the time data, or does it also contain watering data?)

Reply:
So the monolithic fix is to dis-aggregate the "weatherx.py" endpoint into 3 (RemoteIP, Time and Watering). This would also have the advantage of allowing Firmware to call each of those services on a different cycle frequency. I think that would make a good addition for the next Firmware release. But we would also need to maintain backward compatibility so a bit stuck with having a response that returns different things in different circumstances.

In terms of `rawData`, I think we should keep that simple and just use the information returned by `getWateringData()`. At the end of the day, this is the `raw` input to all of the adjustment methods.